### ISSUE TITLE
Fish does not tab complete man pages

### ISSUE THREAD

**Issue by JustinLovinger** [NixOS/nixpkgs#76745]
**Describe the bug**
Tab completion for man pages does not work in the fish shell.

**To Reproduce**
1. Enter a fish shell
2. Type "man " followed by the first few letters of an installed package command
3. Press tab

**Expected behavior**
Fish should complete the man page command.

**Additional context**
This appears related to #14472.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.14, NixOS, 20.03pre204199.3140fa89c51 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(jlove): `""`
 - channels(root): `"nixos-20.03pre204199.3140fa89c51"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: fish, man-db
# a list of nixos modules affected by the problem
module:
```

**Comment by rnhmjoj** [comment]
> This appears related to #14472.

Yes, completion for `man` is implemented by calling `apropos`, which needs a manual cache.

PR #86489 adds a NixOS option to generate the caches but it's not enabled by default: it's a bit expensive and they have to be regenerated every time a new package is installed.

We could enable the caches if `programs.fish.enable` is true, what do you think?

**Comment by JustinLovinger** [comment]
@rnhmjoj I think that makes sense. Fish isn't fully functional without a working `apropos`, and enabling a program should allow it to work to the best of our ability.

**Comment by rnhmjoj** [comment]
> Fish isn't fully functional without a working apropos

All right, once the change lands in master I'll make another PR for fish. Thank you.
